plugin/etcd/plugin.go

```diff
@@ -0,0 +1,451 @@
+	if err != nil {
+		return nil, err
+	}
+
+	prefix, err := endpoint.StringValueDefault("prefix", "")
+	if err != nil {
+		return nil, err
+	}
+
+	return &EtcdConfig{
+		EtcdEndpoints:  etcdEndpoint,
+		Authentication: auth,
+		Username:       user,
+		Password:       pass,
+		ClientCertPath: clientCert,
+		ClientKeyPath:  clientKey,
+		CaCertPath:     caCert,
+		FullOverwrite:  fullOverwrite,
+		Prefix:         prefix,
+	}, nil
+}
+
+func (p EtcdPlugin) Validate(endpoint plugin.ShieldEndpoint) error {
+	var (
+		s    string
+		b    bool
+		err  error
+		fail bool
+	)
+	s, err = endpoint.StringValue("endpoints")
+	if err != nil {
+		fmt.Printf("@R{\u2717 endpoints  %s}\n", err)
+		fail = true
+	} else {
+		fmt.Printf("@G{\u2713 etcd endpoints} data in @C{%s} will be backed up\n", s)
+	}
+
+	b, err = endpoint.BooleanValueDefault("auth", false)
+	if err != nil {
+		fmt.Printf("@R{\u2717 authentication %s}\n", err)
+		fail = true
+	} else if b {
+		fmt.Printf("@G{\u2713 auth} authentication is enabled\n")
+	} else {
+		fmt.Printf("@G{\u2713 auth} authentication is disabled\n")
+	}
+
+	if b {
+		s, err = endpoint.StringValueDefault("user", "")
+		if err != nil {
+			fmt.Printf("@R{\u2717 user  %s}\n", err)
+			fail = true
+		} else if s == "" {
+			fmt.Printf("@G{\u2713 user} username was not provided so cert based auth will be used\n")
+		} else {
+			fmt.Printf("@G{\u2713 username} @C{%s}\n", plugin.Redact(s))
+		}
+
+		s, err = endpoint.StringValueDefault("pass", "")
+		if err != nil {
+			fmt.Printf("@R{\u2717 pass  %s}\n", err)
+			fail = true
+		} else if s == "" {
+			fmt.Printf("@R{\u2713 pass} password was not provided so cert based auth will be used\n")
+		} else {
+			fmt.Printf("@G{\u2713 password} @C{%s}\n", plugin.Redact(s))
+		}
+
+		s, err = endpoint.StringValueDefault("clientCertPath", "")
+		if err != nil {
+			fmt.Printf("@R{\u2717 client certificate path  %s}\n", err)
+		} else if s == "" {
+			fmt.Printf("@R{\u2713 client certificate path} was not provided\n")
+		} else {
+			fmt.Printf("@G{\u2713 client certificate path} was provided\n")
+		}
+
+		s, err = endpoint.StringValueDefault("clientKeyPath", "")
+		if err != nil {
+			fmt.Printf("@R{\u2717 client key path  %s}\n", err)
+		} else if s == "" {
+			fmt.Printf("@R{\u2713 client key path} was not provided\n")
+		} else {
+			fmt.Printf("@G{\u2713 client key path} was provided\n")
+		}
+
+		s, err = endpoint.StringValueDefault("caCertPath", "")
+		if err != nil {
+			fmt.Printf("@R{\u2717 CA certificate path  %s}\n", err)
+		} else if s == "" {
+			fmt.Printf("@R{\u2713 CA certificate path} was not provided\n")
+		} else {
+			fmt.Printf("@G{\u2713 CA certificate path} was provided\n")
+		}
+	}
+
+	b, err = endpoint.BooleanValueDefault("fullOverwrite", false)
+	if err != nil {
+		fmt.Printf("@R{\u2717 full restore  %s}\n", err)
+		fail = true
+	} else if b {
+		fmt.Printf("@G{\u2713} full restore enabled\n")
+	} else {
+		fmt.Printf("@G{\u2713} full restore disabled\n")
```

Comment:
This verbiage is a bit confusing to someone not steeped in how the plugin works.  Consider rewording to explain what overwrite means, and what the scope is.  It's okay to use the value from `prefix` in the messaging here (assuming you validate it beforehand)